### ISSUE TITLE
`CollisionShape3D` size gizmos disappear as soon as refraction is activated on a material behind the shape

### ISSUE THREAD

**Issue by Klaim** [godotengine/godot#93547]
### Tested versions

- spotted in Godot v4.2.2
- still visible in Godot v4.3-beta2

### System information

Godot v4.3.beta2 - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 Ti (NVIDIA; 31.0.15.5222) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

`CollisionShape3D` size gizmos disappear as soon as refraction is activated on a material behind the shape:

![image](https://github.com/godotengine/godot/assets/142265/207d982a-0850-4ece-840e-6e0b8709ff86)

This does not affect CSG size gizmos:
![image](https://github.com/godotengine/godot/assets/142265/1bd26cc3-5093-4f61-bec9-4a36ee6ce71d)

### Steps to reproduce

1. In a 3D scene, create a `CSGBox3D`
2. Add a `StandardMaterial3D` material to that box and enable Refraction in that material
3. In a different position, create either an `Area3D` or `RigidBody3D`
4. Add a child node of type `CollisionShape3D` and set it's shape to `BoxShape3D` or any other shape really
5. Align the editor's camera so that the collision box is in front of the csgbox
6. Select the collision shape -> size gizmos are not visible over the csgbox

### Minimal reproduction project (MRP)

[godot-repro_case-collisionshape3d_gizmo.zip](https://github.com/user-attachments/files/15955693/godot-repro_case-collisionshape3d_gizmo.zip)

**Comment by jsjtxietian** [comment]
Not only CSGBox, but also many others, like raycast:

![image](https://github.com/godotengine/godot/assets/8315986/94c4e22c-1304-4fd9-b32c-4ca221a6e5f9)

**Comment by Klaim** [comment]
Indeed, it's easier to see on 3D volumes and you might not notice depending on the shape and surrounding graphics.
